### PR TITLE
Change G1GC options to match elasticsearch jvm.options

### PR DESCRIPTION
**Pull request by ebadyano**
Relates to https://github.com/elastic/elasticsearch/pull/59667

### CODE REVIEW

File: cars/v1/vanilla/templates/config/jvm.options [hunk 1]
```diff
@@ -39,8 +39,6 @@
 8-13:-XX:CMSInitiatingOccupancyFraction=75
 8-13:-XX:+UseCMSInitiatingOccupancyOnly
 14-:-XX:+UseG1GC
-14-:-XX:G1ReservePercent=25
-14-:-XX:InitiatingHeapOccupancyPercent=30
```
Comment (danielmitterdorfer) [inline cars/v1/vanilla/templates/config/jvm.options:43]: These options are still defined in the case where we enable G1 explicitly (see a few lines below) and I think they need to be removed there as well?

**Review by danielmitterdorfer — CHANGES_REQUESTED** [review]
Thanks! I think we need to remove these options in more places.

File: cars/v1/vanilla/templates/config/jvm.options (hunk 1 above)
Comment (ebadyano) [reply cars/v1/vanilla/templates/config/jvm.options:43]: good catch, thank you! I will push an update

**Review by danielmitterdorfer — APPROVED** [review]
Thanks for iterating. LGTM

**Comment by ebadyano** [comment]
Thank you for the review!

**Comment by danielmitterdorfer** [comment]
@ebadyano given that there is also a backport to 7.x in https://github.com/elastic/elasticsearch/pull/63413 I think this one also needs backporting to `7`?

**Comment by ebadyano** [comment]
> @ebadyano given that there is also a backport to 7.x in [elastic/elasticsearch#63413](https://github.com/elastic/elasticsearch/pull/63413) I think this one also needs backporting to `7`?

@danielmitterdorfer yes, thank you. I was just waiting before I merge the backport on elasticsearch side

**Pull request merged**